certora/specs/Pausable.spec

```diff
@@ -0,0 +1,93 @@
+import "helpers.spec"
+import "methods/IPausable.spec"
+
+methods {
+    pause()
+    unpause()
+}
+
+/*
+┌─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
+│ Function correctness: _pause pauses the contract                                                                    │
+└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+*/
+rule pause(env e) {
+    require nonpayable(e);
+
+    bool pausedBefore = paused();
+
+    pause@withrevert(e);
+    bool success = !lastReverted;
+
+    bool pausedAfter = paused();
+    
+    // liveness
+    assert success => !pausedBefore, "unpause call must succeed if the contract was unpaused before";
+
+    // effect
+    assert success => (pausedBefore != pausedAfter && pausedAfter == true), "contract must be paused after a successful call";
+
+    // no side effect
+    assert !success => (pausedBefore == pausedAfter && pausedAfter == true), "contract must kept paused after a failed call";

```

Comment:
there is no such thing a side effect here. Side effect is if you wanted to check something else beside the paused status. if you fix the liveness, then this becomes irrelevant.

Reply:
I removed it but left the `no side effect` comment for keeping the same format, and explicitly indicating there are no checks.